### ISSUE TITLE
unknown symbolic icons in label page settings ...

### ISSUE THREAD

**Issue by dasnoopy** [Moon-0xff/gnome-mpris-label#99]
[very minor issue) ...as per attached screenshot , in label page setting there are some  unknown symbcolic icons (see filter segments containing and button placeholder settings) - [ extension version  30 - gnome 46 - default adwaita icon theme)

![Screenshot from 2024-04-12 06-46-50](https://github.com/Moon-0xff/gnome-mpris-label/assets/296922/8bd95474-4ae2-4e7a-a08b-3b9e39338a8b)

I'm not a developer but I find out that should be involved this part of code in **pref.js** file.
I replaced the `info-symbolic` icon name with  e.g. `dialog-information-symbolic` and a correct icon appear near the label..

```
function buildInfoButton(labeltooltip){
	let thisInfoButton = new Gtk.MenuButton({
		valign: Gtk.Align.CENTER,
		icon_name: 'info-symbolic',
		visible: true
	});
```

thanks
Andrea

**Comment by Moon-0xff** [comment]
Thanks for reporting this!  
I don't run 46 myself so I'll have to boot up a VM to test (it could be that your system is missing some files!).

**Comment by Batwam** [comment]
Yeah, I have already investigated this as I noticed a similar issue in Gnome nightly. Unfortunately, this icon is not part of Adwaita but of gnome-control-center.

See gitlab issue raised: https://gitlab.gnome.org/GNOME/adwaita-icon-theme/-/issues/283#note_2027550

Do you have gnome control center installed?

When discussing the issue with others, the recommendation was to package the icon as part of the extensions.

**Comment by Moon-0xff** [comment]
I was wondering where the icon was coming from.  
I'm assuming gnome-control-center is or should be packaged alongside gnome.  
Does gnome work well without gnome-control-center?

If not then I think this is the result of a broken package or misconfiguration.

If using gnome without gnome-control-center is a valid use case scenario, then I think we should test if the icon is available, if not, replace it with an icon that's explicitly packaged with gnome, or inside a package that gnome depends on (regardless of distro).

**Comment by Moon-0xff** [comment]
> perhaps that could be a fallback (pretty sure there is a "fallback" method in St.Icon. Would that be a good solution?

In that case it would be a simple one line diff:

```diff
diff --git a/prefs.js b/prefs.js
index 1ee7752..f8c735e 100644
--- a/prefs.js
+++ b/prefs.js
@@ -394,6 +394,7 @@ function buildInfoButton(labeltooltip){
        let thisInfoButton = new Gtk.MenuButton({
                valign: Gtk.Align.CENTER,
                icon_name: 'info-symbolic',
+               fallback_icon_name: 'dialog-information-symbolic',
                visible: true
        });
        thisInfoButton.add_css_class('flat');
```

but `thisInfoButton` is an instance of `Gtk.MenuButton`, not of `St.Icon`, it's worth to try the above diff just in case (I haven't).

**Comment by Batwam** [comment]
you are right, this fallback method doesn't work. I just tried and got: `Error: No property fallback_icon_name on GtkMenuButton`. 

If we want to keep it simple, the easiest is probably to either:
-  switch to `'dialog-information-symbolic'` on the basis that it's an Adwaita icon
-  ignore this on the basis that `gnome-control-center-data` should be present on the system

**Comment by Moon-0xff** [comment]
The commit above will solve this for 45/46, expecting that versions below 45 can always get the icon from `gnome-control-center-data` This might not be the case.

I believe this is a good compromise, as the icon we had was *better*, but apparently problematic for GNOME 46.

**Comment by Batwam** [comment]
Agreed 👍

**Comment by Batwam** [comment]
Issue closed by 02cdae1

**Comment by Moon-0xff** [comment]
> Issue closed by https://github.com/Moon-0xff/gnome-mpris-label/commit/02cdae19cf95649f5c2e6c687bf6848c821e2c0e

Not yet, a new version needs to be uploaded to EGO.
